genai-perf/genai_perf/inputs/converters/rankings_converter.py

```diff
@@ -34,26 +34,49 @@
 class RankingsConverter(BaseConverter):
 
     def convert(self, generic_dataset: GenericDataset, config: InputsConfig) -> Dict:
+        provided_filenames = list(generic_dataset.files_data.keys())
+        if "queries" not in provided_filenames or "passages" not in provided_filenames:
+            raise ValueError(
+                "Both 'queries.jsonl' and 'passages.jsonl' must be present in the input datasets."
+            )
+        
+        queries_data = generic_dataset.files_data["queries"]
+        passages_data = generic_dataset.files_data["passages"]
+
+        rows_of_passage_data = len(passages_data.rows)
+
         request_body: Dict[str, Any] = {"data": []}
 
-        for index, entry in enumerate(generic_dataset["rows"]):
-            model_name = self._select_model_name(config, index)
+        for query_index, query_row in enumerate(queries_data.rows):
+            if query_index > rows_of_passage_data:
```

Comment:
This sets the limit to the number of payloads to be based on the number of passages, in the event that more queries are provided than passages?

Reply:
It actually sets the limit to the min of (number of passage "payloads", number of query "payloads"). I use the term payloads because there are usually multiple passages per query. So it's based on rows, not the number of texts.

This avoids the need for a strict check that passage_rows == query_rows and instead uses the maximum usable subset in the case where the number differs, which should be uncommon.